### ISSUE TITLE
`esModuleInterop` breaks circular imports

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/TypeScript#33012]
**TypeScript Version:**  3.5.2

**Search Terms:**

**Code**

`./tsconfig.json`:

``` json
{
    "compilerOptions": {
        "esModuleInterop": true
    },
    "files": ["./src/main.ts"]
}
```

`./src/main.ts`

```ts
import { other } from './other';

console.log(other.foo1);
```

`./src/other.ts`

```ts
export const foo1 = 1;
export const foo2 = 2;

// Import *this* module as a namespace
// Unexpected error:
// Circular definition of import alias 'other'
import * as other from './other';

// Re-export the namespace as a named export
export { other };
```

(For context on why you might do this: [it allows you to give namespace imports a common name](https://medium.com/unsplash/named-namespace-imports-7345212bbffb).)

**Expected behavior:**

No error

**Actual behavior:**

Error (see code comment)
